type-check.ts

```diff
@@ -413,6 +413,19 @@ export function tcExpr(env: GlobalTypeEnv, locals: LocalTypeEnv, expr: Expr<null
       } else {
         throw new TypeCheckError("method calls require an object");
       }
+    case "bracket-lookup":
+      var obj_t = tcExpr(env, locals, expr.obj);
+      var key_t = tcExpr(env, locals, expr.key);
+      var tBracketExpr = { ...expr, obj: obj_t, key: key_t, a: obj_t.a };
+      if (obj_t.a != STRING) {
+        throw new TypeCheckError("Bracket lookup on " + obj_t.a.tag + " type not possible");
+      }
+      if (key_t.a != NUM) {
```

Comment:
You might want to use a type comparison helper function here.

Reply:
Thanks for the suggestion!